test/libsolidity/semanticTests/libraries/library_return_struct_with_mapping.sol

```diff
@@ -0,0 +1,20 @@
+pragma experimental ABIEncoderV2;
+
+library Lib {
+    struct Items {
+        mapping (uint => uint) a;
+    }
+
+    function get() public returns (Items storage x) {
+        assembly { x.slot := 123 }
+    }
+}
+
+contract C {
+    function f() public returns(uint256 slot) {
+        Lib.Items storage ptr = Lib.get();
+        assembly { slot := ptr.slot }
+    }
+}
```

Comment:
Compiles now, but test fails...

Reply:
How does it fail?

Reply:
I think some `revert()` or `invalid()` is happening during `Lib.Items storage ptr = Lib.get();`